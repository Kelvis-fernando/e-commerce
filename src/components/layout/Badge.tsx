import { ReactElement } from "react";
import { BadgeContainer } from "../../styles/layout/badge";

interface BadgeProps {
  children: ReactElement;
  items: number;
}
const Badge = ({ children, items }: BadgeProps) => {
  return (
    <BadgeContainer>
      <div className="badge">{items}</div>
      {children}
    </BadgeContainer>
  );
};

export default Badge;
