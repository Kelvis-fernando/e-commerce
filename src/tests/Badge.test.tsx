import React from "react";
import { render } from "@testing-library/react";
import Badge from "../components/layout/Badge";

describe("Badge", () => {
  test("should render and match snapshot", () => {
    const { container } = render(
      <Badge items={0}>
        <p>test</p>
      </Badge>
    );

    expect(container).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
