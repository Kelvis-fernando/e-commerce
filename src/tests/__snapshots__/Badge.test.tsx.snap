// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render and match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe gIKsQF"
  >
    <p>
      test
    </p>
  </div>
</div>
`;
